predelete/predelete.go

```diff
@@ -62,6 +67,33 @@ func PreDeleteOperatorSecret(clientset *kubernetes.Clientset) error {
 	return nil
 }
 
+func PreDeleteEnterprisePortalSecret(clientset *kubernetes.Clientset) error {
+	fmt.Println("Running pre delete hook")
+	secrets, err := clientset.CoreV1().Secrets(os.Getenv("TYK_POD_NAMESPACE")).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	found := false
+	for _, value := range secrets.Items {
+		if value.Name == os.Getenv("ENTERPRISE_PORTAL_SECRET_NAME") {
+			err = clientset.CoreV1().Secrets(os.Getenv("TYK_POD_NAMESPACE")).Delete(context.TODO(), value.Name, metav1.DeleteOptions{})
+			if err != nil {
+				return err
+			}
+			found = true
+			break
+		}
+	}
+
+	if found == false {
+		fmt.Println("A previously created enterprise portal secret has not been identified")
```

Comment:
Am I correct that users will see it on their first run? isn't it a bit of an odd message to get on the first run? 
Maybe try to rephrase it? "A previously created enterprise" feels like a complicated sentense

Reply:
No this is only viewed on the predelete action.